File: _src/06/Chapter6.CSharp/Models/Person.cs
﻿using System;
using System.Collections.Generic;

namespace Chapter6.CSharp.Models
{
    public class Person
    {
        public Person()
        {
            Phones = new HashSet<Phone>();
            Companies = new HashSet<Company>();
        }
        public int PersonId { get; set; }
        public int? PersonTypeId { get; set; }
        public virtual PersonType PersonType { get; set; }
        public string FirstName { get; set; }
        public string LastName { get; set; }
        public DateTime? BirthDate { get; set; }
        public decimal HeightInFeet { get; set; }
        public byte[] RowVersion { get; set; }
        public bool IsActive { get; set; }
        public virtual ICollection<Phone> Phones { get; set; }
        public ICollection<Company> Companies { get; set; }
    }

}

File: _src/06/Chapter6.VB/Models/Person.vb


Namespace Models
    Public Class Person
        Public Sub New()
            Phones = New HashSet(Of Phone)
            Companies = New HashSet(Of Company)
        End Sub

        Property PersonId() As Integer
        Property PersonTypeId() As Integer?
        Overridable Property PersonType() As PersonType
        Property FirstName() As String
        Property LastName() As String
        Property BirthDate() As DateTime?
        Property HeightInFeet() As Decimal
        Property RowVersion() As Byte()
        Property IsActive() As Boolean
        Overridable Property Phones() As ICollection(Of Phone)
        Overridable Property Companies() As ICollection(Of Company)
    End Class

End Namespace